test/boilerplate_suggestors/props_mixin_migrator_test.dart

```diff
@@ -12,22 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import 'dart:convert';
-
 import 'package:over_react_codemod/src/boilerplate_suggestors/boilerplate_utilities.dart';
 import 'package:over_react_codemod/src/boilerplate_suggestors/props_mixins_migrator.dart';
 import 'package:test/test.dart';
 
 import '../util.dart';
-import 'boilerplate_utilities_test.dart';
 
-main() {
+void main() {
+  group('PropsMixinMigrator', () {
+    group('', () {
+      PropsMixinMigratorTestHelper(
+        path: 'test/boilerplate_suggestors/semver_report.json',
+        shouldTreatAllComponentsAsPrivate: false,
+        isValidFilePath: true,
+      );
+    });
+
+    group('with --treat-all-components-as-private flag', () {
+      PropsMixinMigratorTestHelper(
+        path: 'test/boilerplate_suggestors/semver_report.json',
+        shouldTreatAllComponentsAsPrivate: true,
+        isValidFilePath: true,
+      );
+    });
+
+    group('with invalid file path', () {
+      PropsMixinMigratorTestHelper(
+        path: 'test/boilerplate_suggestors/does_not_exist.json',
+        shouldTreatAllComponentsAsPrivate: false,
+        isValidFilePath: false,
+      );
+    });
+  });
+}
+
+PropsMixinMigratorTestHelper({
+  String path,
+  bool shouldTreatAllComponentsAsPrivate,
+  bool isValidFilePath,
+}) {
   group('PropsMixinMigrator', () {
     final converter = ClassToMixinConverter();
-    final testSuggestor = getSuggestorTester(PropsMixinMigrator(converter));
+    final semverHelper = getSemverHelper(path,
+        shouldTreatAllComponentsAsPrivate: shouldTreatAllComponentsAsPrivate);
+    SuggestorTester testSuggestor;
 
-    setUpAll(() {
-      semverHelper = SemverHelper(jsonDecode(reportJson));
+    setUpAll(() async {
+      testSuggestor =
+          getSuggestorTester(PropsMixinMigrator(converter, await semverHelper));
```

Comment:
I also noted above about the usage of `await` here versus in the instantiation - so just noting it here as well